File: ThematicMapCreator.Api/Contracts/LayerStyleOptions/DensityMapStyleOptions.cs
using System;

namespace ThematicMapCreator.Api.Contracts.LayerStyleOptions
{
    public class DensityMapStyleOptions : ILayerStyleOptions
    {
        public string Color { get; set; }
        public string FillColor { get; set; }
        public double MaxValue { get; set; }
        public double MinValue { get; set; }
        public string PropertyName { get; set; }
        public double Size { get; set; }
        public LayerStyle Style { get; } = LayerStyle.DensityMap;
    }
}

File: ThematicMapCreator.Client/src/app/models/layer-style-options/density-map-style-options.ts
import { LayerStyleOptions } from './layer-style-options';
import { LayerStyle } from './layer-style.enum';

export class DensityMapStyleOptions implements LayerStyleOptions {
    style: LayerStyle.DensityMap;
    size = 8;
    color = '#000000';
    fillColor = '#000000';
    propertyName: string;
    minValue: number;
    maxValue: number;
}
